jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/AuthParameters.java

```diff
@@ -16,22 +16,48 @@
 
 package com.google.cloud.tools.jib.gradle;
 
+import com.google.cloud.tools.jib.plugins.common.AuthProperty;
+import com.google.common.base.Preconditions;
 import javax.annotation.Nullable;
 import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
 import org.gradle.api.tasks.Optional;
 
 /**
  * A bean that configures authorization credentials to be used for a registry. This is configurable
  * with Groovy closures and can be validated when used as a task input.
  */
-public class AuthParameters {
+public class AuthParameters implements AuthProperty {
 
   @Nullable private String username;
   @Nullable private String password;
+  @Nullable private String usernameDescriptor;
+  @Nullable private String passwordDescriptor;
+
+  @Internal
+  @Override
+  public String getUsernamePropertyDescriptor() {
+    return Preconditions.checkNotNull(usernameDescriptor);
+  }
+
+  void setUsernamePropertyDescriptor(String usernameDescriptor) {
```

Comment:
These could be set via a constructor. (`ObjectFactory#newInstance` takes constructor parameters)